src/nvim/option.c

```diff
@@ -4667,7 +4664,9 @@ char_u *check_colorcolumn(win_T *wp)
       col = col * getdigits_int(&s);
       if (wp->w_buffer->b_p_tw == 0)
         goto skip;          /* 'textwidth' not set, skip this item */
-      col += wp->w_buffer->b_p_tw;
+      assert(wp->w_buffer->b_p_tw >= (long)INT_MIN - col
+             && wp->w_buffer->b_p_tw <= (long)INT_MAX - col);
```

Comment:
This won't work when sizeof(long) == sizeof(int).
Following the same spirit as before, I could change it to:
```
       assert((col >= 0                                                           
               && wp->w_buffer->b_p_tw <= INT_MAX - col                           
               && wp->w_buffer->b_p_tw + col >= INT_MIN)                          
              || (col < 0                                                         
                  && wp->w_buffer->b_p_tw >= INT_MIN - col                        
                  && wp->w_buffer->b_p_tw + col <= INT_MAX));
```
But I've found it's easier if we test the opposite (not overflow condition instead of in-bounds condition):
```
      assert(!(col > 0 && wp->w_buffer->b_p_tw > INT_MAX - col                   
                || col < 0 && wp->w_buffer->b_p_tw < INT_MIN - col));             
```
I will review other asserts, to follow this shorter (and IMO, clearer) style.